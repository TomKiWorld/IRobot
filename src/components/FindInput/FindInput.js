import React from 'react';

const FindInput = ({ onNameChange }) => {
  return (
    <div className='pa2'>
      <input
        aria-label='set-name'
        className='pa3 ba b--green bg-lightest-blue'
        type='search'
        placeholder='Type your name'
        onChange={(event) => onNameChange(event.target.value)}
      />
    </div>
  );
}

export default FindInput;
