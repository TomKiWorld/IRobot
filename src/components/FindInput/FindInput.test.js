import { shallow } from 'enzyme';
import React from 'react';
import FindInput from './FindInput';

describe('FindInput Component', () => {
  const mockState = { name: '' };
  const onChange= (text) => {
    mockState.name = text;
  }
  const wrapper = shallow(<FindInput onNameChange={onChange}/>);
  it('Expect to render FindInput Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  const searchInput = wrapper.find('[type=\'search\']').at(0);
  searchInput.simulate('change', { target: { value: 'Name'} });
  it('Expect value to change', () => {
    expect(mockState.name).toEqual('Name');
  });
});
