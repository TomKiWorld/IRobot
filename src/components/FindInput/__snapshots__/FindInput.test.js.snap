// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindInput Component Expect to render FindInput Component 1`] = `ShallowWrapper {}`;
